### PR TITLE
Inline preview/ghost text screen reader a11y

### PR DESCRIPTION
**Pull request by akoreman**
*Issue #, if available:* NA

*Description of changes:* Currently, there is no way for screen reader users to learn the contents of the ghost text inline preview. This adds an `aria-describedby` to the completion item with inline preview which points to a screen reader only div displaying the contents of the ghost test. This allows the screen reader to read the content of the inline preview using the `aria-describedby` UX of their screen reader.

https://github.com/ajaxorg/ace/assets/34573235/d95f53dc-0aad-448e-84c1-3c4c2ec279da

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by akoreman** [comment]
Switched the implementation from using `aria-details` to `aria-describedby` due to the [lacking support](https://a11ysupport.io/tech/aria/aria-details_attribute) for `aria-details`.

File: src/autocomplete/inline_screenreader.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            this.screenReaderDiv.removeChild(this.screenReaderDiv.firstChild);
+        }
+        this._lines = content.split(/\r\n|\r|\n/);
+        const codeElement = this.createCodeBlock();
+        this.screenReaderDiv.appendChild(codeElement);
+    }
+
+    destroy() {
+        // Remove all children of the div
+        while (this.screenReaderDiv.firstChild) {
```
Comment (whazor) [inline src/autocomplete/inline_screenreader.js:55]: Is this needed when you remove the div itself below?

Comment (akoreman) [reply src/autocomplete/inline_screenreader.js:55]: What happens if you remove an element which has child nodes? (I was expecting bad things to happen, which is why I removed the child nodes first here)

Comment (whazor) [reply src/autocomplete/inline_screenreader.js:55]: They will be removed as well.

File: src/autocomplete_test.js [hunk 2]
```diff
@@ -778,6 +778,48 @@ module.exports = {
...
+        // Popup should be open, with inline text renderered.
+        assert.equal(editor.completer.popup.isOpen, true);  
+        assert.equal(completer.popup.getRow(), 0);
+        assert.strictEqual(inline.isOpen(), true);
+        assert.strictEqual(editor.renderer.$ghostText.text, "function\nthat does something\ncool");
+
+        editor.completer.popup.renderer.$loop._flush();
+        var popupTextLayer = completer.popup.renderer.$textLayer;
+
+        // aria-details of selected popup item should have aria-details set to the offscreen inline screen reader div.
```
Comment (InspiredGuy) [inline src/autocomplete_test.js:814]: nit: it's not `aria-details` anymore, it's `aria-describedby`

File: src/autocomplete/inline_screenreader.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    let idString = "doc-tooltip ";
+                    for (let lineIndex = 0; lineIndex < this._lines.length; lineIndex++) {
+                        idString += `ace-inline-screenreader-line-${lineIndex} `;
+                    }
+                    selected.setAttribute("aria-describedby", idString);      
+                }
+            }.bind(this));
+        }
+
+        // TODO: Path for when special inline completion popup is used.
```
Comment (InspiredGuy) [inline src/autocomplete/inline_screenreader.js:42]: is there a `special` inline completion popup?

File: src/autocomplete_test.js (hunk 2 above)
Comment (akoreman) [reply src/autocomplete_test.js:814]: good catch, let me fix this

File: src/autocomplete/inline_screenreader.js (hunk 3 above)
Comment (InspiredGuy) [reply src/autocomplete/inline_screenreader.js:42]: do we need to handle this case as part of this PR?

File: src/autocomplete_test.js (hunk 2 above)
Comment (akoreman) [reply src/autocomplete_test.js]: fixed this

File: src/autocomplete/inline_screenreader.js (hunk 3 above)
Comment (akoreman) [reply src/autocomplete/inline_screenreader.js:42]: Added GH issue to todo note in code.

**Pull request merged**